scripts/login-page.nse

```diff
@@ -0,0 +1,155 @@
+--
+--  If timeout occurs frequently due to bad internet connection then
+--  @usage ./nmap --script login-page --script-args extension="php" <target> --host-timeout=<timeout> -d
+--
+--  Best way to run the script
+--  If the user has prior knowledge on which port to check, he can save time by
+--  specifying that particular port as a general command line argument using -p
+--  @usage ./nmap --script login-page --script-args extension="jsp" <target> -p 80 -d
+--
+--  @args login-page.extension Checks for pages of particular extension,
+--        default is extension is all which checks for all the extensions.
+--
+--  @output
+--  PORT   STATE SERVICE REASON
+--  22/tcp open  ssh     syn-ack ttl 64
+--  80/tcp open  http    syn-ack ttl 64
+--  | login-page:
+--  |   192.168.146.145/admin/
+--  |   192.168.146.145/admin/index.php
+--  |_  192.168.146.145/admin/login.php
+---
+
+author = "Rewanth Cool"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"discovery", "safe"}
+
+portrule = function(host, port)
+  -- Working :
+  -- If -p argument is used, then the script will be executed only on the given ports.
+  -- If -p argument is not used, then the script will be executed on all the open ports.
+  stdnse.debug(string.format("Loading port %d", port.number))
+
+  local selected_ports = shortport.port_or_service({port.number}, "http", "tcp", {"open", "open|filtered"})
+  return selected_ports(host, port)
+end
+
+action = function(host, port)
+
+  local path = "/"
+  local extension = stdnse.get_script_args(SCRIPT_NAME .. ".extension") or "all"
+
+  --  NOTE:
+  --  If any new db file is created for non-existing extension
+  --  make sure to update the name of the extension here also
+  --  inorder to execute this script successfully.
+  local existing_extensions = {
+    "aspx",
+    "asp",
+    "brf",
+    "cgi",
+    "cfm",
+    "js",
+    "jsp",
+    "php"
+  }
+
+  -- Checking whether the user provided extension is existing in our db or not.
+  local flag = 0
+  local string_of_existing_extensions = ''
+  for _, v in ipairs(existing_extensions) do
+    string_of_existing_extensions = string_of_existing_extensions .. v .. ', '
+    if extension == v then
+      flag = 1
+    end
+  end
+
+  string_of_existing_extensions = string_of_existing_extensions .. 'all'
+
+  -- If user provided extension is not in our db
+  if flag == 0 then
+    -- Extension type not found in the db, throwing a suggestion to end user.
+    local err = "There is no extension like '" .. extension .. "' in the db."
+    err = err .. " Send a mail to rewanth1997[at]gmail.com"
+    err = err .. " and it will be updated soon to db."
+    err = err .. " Available extensions are "
+    err = err .. string_of_existing_extensions .. "\n"
+    return err
+  end
+
+  local file = "nselib/data/web-login/" .. extension .. ".lst"
+
+  -- These regex are used while scraping the web pages for confirmation.
+  local regex = {

```

Comment:
I guess all the websites use only Password, password or PASSWORD. I thought this will be the standard way of coding and I mentioned them exclusively. I don't think anyone will use these kinds of words on their websites pASSwoRd, PASSwORD.

But still your suggestion have the power to improve the accuracy of results, I will implement the case insensitive search also soon.